draft-ietf-oauth-identity-assertion-authz-grant.md

```diff
@@ -217,69 +220,37 @@ The example flow is for an enterprise `acme`, which uses a multi-tenant wiki app
+2. Client uses the Access Token or Refresh Token to request an Identity Assertion JWT Authorization Grant for the Resource Authorization Server from the Enterprise Authorization Server
 3. Client exchanges the Identity Assertion JWT Authorization Grant for an Access Token at the Resource Authorization Server's token endpoint
 4. Client makes an API request to the Resource Server with the Access Token
 
-This specification is constrained to deployments where a set of Resource Authorization Servers for applications used by an organization are trusting the same IdP Authorization Server for Single Sign-On (SSO). The IdP Authorization Server provides a consistent trust boundary and user identity for the set of Resource Authorization Servers to honor the ID-JAG issued by the IdP.  The Resource Authorization Server not only delegates user authentication but also delegates user authorization authority to the IdP Authorization Server for the scopes and resource specified in the ID-JAG and does not need obtain user consent directly from the resource owner.
+This specification is constrained to deployments where a set of Resource Authorization Servers for applications used by an organization are trusting the same IdP for Single Sign-On (SSO), and the same Enterprise Authorization Server for authorization. The IdP provides a consistent trust boundary and user identity for the set of Resource Authorization Servers to honor the ID-JAG issued by the Enterprise Authorization Server.  The Resource Authorization Server not only delegates user authentication but also delegates user authorization authority to the Enterprise Authorization Server for the scopes and resource specified in the ID-JAG and does not need obtain user consent directly from the resource owner. It is also possible for the Enterprise Authorization Server to be built into the IdP itself.
 
 
 ## User Authentication
 
-The Client initiates an authentication request with the IdP Authorization Server using OpenID Connect or SAML.
+The Client initiates an authentication request with the IdP using OpenID Connect or SAML.
 
 The following is an example using OpenID Connect
 
     302 Redirect
     Location: https://acme.idp.example/authorize?response_type=code&scope=openid%20offline_access&client_id=...
 
-The user authenticates with the IdP, and is redirected back to the Client with an authorization code, which it can then exchange for an ID Token and optionally a Refresh Token when `offline_access` scope is requested per {{OpenID.Core}}.
+The user authenticates with the IdP, and is redirected back to the Client with an authorization code, which it can then exchange for an ID Token, Access Token, and optionally a Refresh Token when `offline_access` scope is requested per {{OpenID.Core}}.
```

Comment:
See comment above about access token needed to be requested for a different resource than the IdP.  This may cause challenges if RPs want to also use the access token to access UserInfo or other protected resources at the IdP (e.g. Graph API or some other discovery resource).  

We also need to make some statement about scopes for the access token.  Are we going to leave this to be implementation specific (e.g. Enterprise AS can trust whatever scopes it wants and the Client needs to know what magic scope to request) or scope standardized by this spec or there is a discovery mechanism for the Client to know what scope to request

Reply:
Agree with the `aud` concern in the Access Token. 

It might be good to specify the scope that enables the EAS to issue ID-JAGs, but on the other hand, it could be left as an implementation concern.

Reply:
Given that OpenID Connect defines scopes used for SSO, it probably does make sense to define scopes for this.

It's actually the default to have [multiple values](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.3) in the `aud` claim, and I don't see anything wrong with issuing an access token that is accepted by the IdP for the userinfo endpoint and accepted by the EAS for this use.

Reply:
I'm OK with multiple audiences for access token just think this may need to be clear to reader as it puts some requirements on the IdP (e.g not all IdPs may support multiple audiences) that may affect client interoperability (when can I make a single authorization request vs when do I need to split them out).